gtk4/sample/examples/application7/exampleapp.rb

```diff
@@ -0,0 +1,307 @@
+          <property name="default-width">600</property>
+          <property name="default-height">400</property>
+          <child type="titlebar">
+            <object class="GtkHeaderBar" id="header">
+              <child type="title">
+                <object class="GtkStackSwitcher" id="tabs">
+                  <property name="stack">stack</property>
+                </object>
+              </child>
+              <child type="end">
+                <object class="GtkMenuButton" id="gears">
+                  <property name="direction">none</property>
+                </object>
+              </child>
+              <child type="end">
+                <object class="GtkToggleButton" id="search">
+                  <property name="sensitive">0</property>
+                  <property name="icon-name">edit-find-symbolic</property>
+                </object>
+              </child>
+            </object>
+          </child>
+          <child>
+            <object class="GtkBox" id="content_box">
+              <property name="orientation">vertical</property>
+              <child>
+                <object class="GtkSearchBar" id="searchbar">
+                  <child>
+                    <object class="GtkSearchEntry" id="searchentry">
+                      <signal name="search-changed" handler="search_text_changed"/>
+                    </object>
+                  </child>
+                </object>
+              </child>
+              <child>
+                <object class="GtkStack" id="stack">
+                  <signal name="notify::visible-child" handler="visible_child_changed"/>
+                </object>
+              </child>
+            </object>
+          </child>
+        </template>
+      </interface>
+      TEMPLATE
+      set_template(data: template)
+      bind_template_child("stack")
+      bind_template_child("gears")
+      bind_template_child("search")
+      bind_template_child("searchbar")
+      bind_template_callback_full("search_text_changed")
+      bind_template_callback_full("visible_child_changed")
+    end
+  end
+
+  def initialize(application)
+    menu_ui = <<~MENU
+    <?xml version="1.0" encoding="UTF-8"?>
+    <interface>
+      <menu id="menu">
+        <section>
+          <item>
+            <attribute name="label" translatable="yes">_Preferences</attribute>
+            <attribute name="action">app.preferences</attribute>
+          </item>
+        </section>
+        <section>
+          <item>
+            <attribute name="label" translatable="yes">_Quit</attribute>
+            <attribute name="action">app.quit</attribute>
+          </item>
+        </section>
+      </menu>
+    </interface>
+    MENU
+    super(application: application)
+    builder = Gtk::Builder.new(string: menu_ui)
+    gears.menu_model = builder["menu"]
+    @settings = Gio::Settings.new("org.gtk.exampleapp")
+    @settings.bind("transition", stack, "transition-type", Gio::SettingsBindFlags::DEFAULT)
```

Comment:
Can we use `:default` here?

```suggestion
    @settings.bind("transition", stack, "transition-type", :default)
```